src/mcts/node.cc

```diff
@@ -376,11 +376,23 @@ V4TrainingData Node::GetV4TrainingData(GameResult game_result,
 
   // Aggregate evaluation Q.
   result.root_q = -GetQ();
+  if (result.root_q < -1 || result.root_q > 1) { 
```

Comment:
Maybe add a helper function `Check` or `ExceptionUnless` into `utils/exception.h`?

And then call `ExceptionUnless(-1 <= result.root_q  && result.root_q <= 1, "Corrupt root q");` or even `Check(-1 <= result.root_q  &&| result.root_q <= 1) << "Corrupt root q " << result.root_q;` ?

Then it won't look like it's golang. :) And if it's macro like we have for CERR in logging.h, we can also log filename and line.